dotnet/src/SemanticKernel.UnitTests/PromptTemplate/PromptTemplateConfigTests.cs

```diff
@@ -141,6 +142,108 @@ public void DeserializingExpectCompletion()
+                    "type": "auto",
+                    "functions":["p1.f1"]
+                  }
+                }
+              }
+            }
+            """;
+
+        // Act
+        var promptTemplateConfig = PromptTemplateConfig.FromJson(configPayload);
+
+        // Assert
+        Assert.NotNull(promptTemplateConfig);
+        Assert.Single(promptTemplateConfig.ExecutionSettings);
+
+        var executionSettings = promptTemplateConfig.ExecutionSettings.Single().Value;
+
+        var autoFunctionCallChoice = executionSettings.FunctionChoiceBehavior as AutoFunctionChoiceBehavior;
+        Assert.NotNull(autoFunctionCallChoice);
+
+        Assert.NotNull(autoFunctionCallChoice.Functions);
+        Assert.Equal("p1-f1", autoFunctionCallChoice.Functions.Single());
+    }
+
+    [Fact]
+    public void DeserializingRequiredFunctionCallingChoice()
+    {
+        // Arrange
+        string configPayload = """
+            {
+              "schema": 1,
+              "execution_settings": {
+                "default": {
+                  "model_id": "gpt-4",
+                  "function_choice_behavior": {
+                    "type": "required",
+                    "functions":["p1.f1"]
+                  }
+                }
+              }
+            }
+            """;
+
+        // Act
+        var promptTemplateConfig = PromptTemplateConfig.FromJson(configPayload);
+
+        // Assert
+        Assert.NotNull(promptTemplateConfig);
+        Assert.Single(promptTemplateConfig.ExecutionSettings);
+
+        var executionSettings = promptTemplateConfig.ExecutionSettings.Single().Value;
+        Assert.NotNull(executionSettings);
+
+        var requiredFunctionCallChoice = executionSettings.FunctionChoiceBehavior as RequiredFunctionChoiceBehavior;
+        Assert.NotNull(requiredFunctionCallChoice);
+
+        Assert.NotNull(requiredFunctionCallChoice.Functions);
+        Assert.Equal("p1-f1", requiredFunctionCallChoice.Functions.Single());
+    }
+
+    [Fact]
+    public void DeserializingNoneFunctionCallingChoice()
+    {
+        // Arrange
+        string configPayload = """
+            {
+              "schema": 1,
+              "execution_settings": {
+                "default": {
+                  "model_id": "gpt-4",
+                  "function_choice_behavior": {
+                    "type": "none"
+                  }
+                }
+              }
+            }
+            """;
+
+        // Act
+        var promptTemplateConfig = PromptTemplateConfig.FromJson(configPayload);
+
+        // Assert
+        Assert.NotNull(promptTemplateConfig);
+        Assert.Single(promptTemplateConfig.ExecutionSettings);
+
+        var executionSettings = promptTemplateConfig.ExecutionSettings.Single().Value;
+
+        var noneFunctionCallChoice = executionSettings.FunctionChoiceBehavior as NoneFunctionChoiceBehavior;
+        Assert.NotNull(noneFunctionCallChoice);
+    }
+
```

Comment:
I assume you'll be updating the PR to handle the abstraction on at least Mistral and Gemini as well?

Reply:
I planned to update both in separate PRs to minimize the PR scope.

Reply:
The abstraction will be applied to Gemini and mistral in scope of the issues:
- https://github.com/microsoft/semantic-kernel/issues/6555
- https://github.com/microsoft/semantic-kernel/issues/5953